rpc_jobs/rpc_aio.yml

```diff
@@ -61,27 +66,23 @@
           branches: "do_not_build_on_pr"
           NUM_TO_KEEP: 10
     exclude:
+      # Upgrade jobs are only run for mitaka.
       - series: liberty
-        context: upgrade
+        action: majorupgrade
       - series: newton
-        context: upgrade
-      - series: artifacts
-        context: upgrade
+        action: majorupgrade
       - series: master
-        context: upgrade
+        action: majorupgrade
       # Xenial builds are run for newton and above.
       - series: liberty
-        context: xenial
+        image: xenial
       - series: mitaka
-        context: xenial
-      # Artifacts are currently only built for Trusty
-      - series: artifacts
-        context: xenial
-      # An artifacted ceph deployment is not yet supported
-      - series: artifacts
-        context: ceph
+        image: xenial
+      # Ceph builds are only run for trusty
+      - image: xenial
+        scenario: ceph
     jobs:
-      - 'RPC-AIO_{series}-{context}-{ztrigger}'
+      - 'RPC-AIO_{series}-{image}-{scenario}-{action}-{ztrigger}'
 
 - job-template:
     # DEFAULTS
```

Comment:
If we are changing this, I'd say stages should be part of the action. Every different action will have different stages. 

Or alternatively, no job should have different stages, and we handle this in a gating script that's called.